### PR TITLE
implement rerun functionality

### PR DESCRIPTION
**Pull request by rplevka**
simply add `--rp-rerun` with or without `--rp-rerun-of xxxx-xxxx-xxxx-xxxx` to indicate the current session should be a rerun of the recent or specific launch

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -87,6 +87,8 @@ The following parameters are optional:
 - :code:`rp_launch_description = 'Smoke test'` - launch description (could be overridden
   by pytest --rp-launch-description option, default value is '')
 
+- :code:`rp_rerun = True` - treat the Launch as a rerun of the recent Launch of the same name
+- :code:`rp_rerun_of = xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx` - treat the Launch as a rerun of the given Launch (to be used together with rp_rerun = True)
```
Comment (iivanou) [inline README.rst:91]: According to the [dev guides](https://github.com/reportportal/documentation/blob/master/src/md/src/DevGuides/reporting.md#start-launch), `rerun_of` and `rerun` should be used together. So, in our case, it will be enough to only specify `rp_rerun_of` with the launch ID to inform the agent that we want to rerun an existing launch.

File: pytest_reportportal/service.py [hunk 2]
```diff
@@ -161,7 +163,9 @@ def init_service(self,
                 log_batch_size=log_batch_size,
                 retries=retries,
                 verify_ssl=verify_ssl,
-                launch_id=custom_launch
+                launch_id=custom_launch,
+                rerun=rerun,
```
Comment (iivanou) [inline pytest_reportportal/service.py:167]: This option is not processed in the constructor of the client.

File: pytest_reportportal/service.py [hunk 3]
```diff
@@ -161,7 +163,9 @@ def init_service(self,
...
+                rerunof=rerun_of
```
Comment (iivanou) [inline pytest_reportportal/service.py:168]: This option is not processed in the constructor of the client.

File: pytest_reportportal/plugin.py [hunk 4]
```diff
@@ -78,6 +78,8 @@ def pytest_sessionstart(session):
                 log_batch_size=int(session.config.getini('rp_log_batch_size')),
                 ignore_errors=bool(session.config.getini('rp_ignore_errors')),
                 custom_launch=session.config.getini('rp_launch_id') or None,
+                rerun=session.config.getini('rp_rerun') or False,
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:81]: No need to pass these options to the init_service method. There are being processed in the start_launch only.

File: pytest_reportportal/plugin.py [hunk 5]
```diff
@@ -188,7 +192,10 @@ def pytest_configure(config):
     if not config.option.rp_launch_description:
         config.option.rp_launch_description = config.\
             getini('rp_launch_description')
-
+    if not config.option.rp_rerun:
+        config.option.rp_rerun = config.getini('rp_rerun')
+    if not config.option.rp_rerun_of:
+        config.option.rp_rerun_of = config.getini('rp_rerun_of')
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:198]:
```suggestion
        config.option.rp_rerun_of = config.getoption('rp_rerun_of', default=None)
        config.option.rp_rerun = bool(config.option.rp_rerun_of)
```

**Review by iivanou — CHANGES_REQUESTED** [review]

File: README.rst (hunk 1 above)
Comment (rplevka) [reply README.rst:91]: fair enough.

File: pytest_reportportal/plugin.py (hunk 5 above)
Comment (rplevka) [reply pytest_reportportal/plugin.py:198]: this won't work correctly - it is possible to use `rp_rerun` without `rp_rerun_of` (in which case the recent matching launch is used). your suggestion would disable such functionality.

Also, i don't get this:
```
config.option.rp_rerun_of = config.getoption('rp_rerun_of', default=None)
```
If i can't see the `rp_rerun_of` in inline options, i want to check the ini file,....i don't think your suggestion does that. I'd assume this was a typo, but `config.getini` [does not support "default"](https://docs.pytest.org/en/reorganize-docs/writing_plugins.html#_pytest.config.Config.getini)

Comment (rplevka) [reply pytest_reportportal/plugin.py:198]: I believe this perfectly covers it all, including your above comment (implicit `rp_rerun = True` if `rp_rerun_of is not None`):

```
    if not config.option.rp_rerun_of:
        config.option.rp_rerun_of = config.getini('rp_rerun_of')
    if config.option.rp_rerun_of:
        config.option.rp_rerun = True
    else:
        if not config.option.rp_rerun:
            config.option.rp_rerun = config.getini('rp_rerun')
```

**Comment by rplevka** [comment]
Comments addressed, thanks for your review

File: pytest_reportportal/plugin.py (hunk 5 above)
Comment (iivanou) [reply pytest_reportportal/plugin.py:193]: > it is possible to use `rp_rerun` without `rp_rerun_of` (in which case the recent matching launch is used...

Where did you find this information? I'm not sure. It is missing in the documentation.

Comment (rplevka) [reply pytest_reportportal/plugin.py:193]: https://github.com/reportportal/documentation/blob/master/src/md/src/DevGuides/rerun.md#latest-launch

Comment (iivanou) [reply pytest_reportportal/plugin.py:193]: Ok, I have not seen that doc before...

File: pytest_reportportal/plugin.py [hunk 6]
```diff
@@ -183,12 +185,18 @@ def pytest_configure(config):
     if config._reportportal_configured is False:
         return
 
-    if not config.option.rp_launch:
-        config.option.rp_launch = config.getini('rp_launch')
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:187]: Why did you remove this?

**Review by iivanou — CHANGES_REQUESTED** [review]
Tests are failing...

File: pytest_reportportal/plugin.py (hunk 6 above)
Comment (rplevka) [reply pytest_reportportal/plugin.py:187]: that happened by a mistake, i'm sorry

**Comment by rplevka** [comment]
@iivanou - flake8 errors fixed and the rp_launch option parsing returned to its place.

**Review by iivanou — APPROVED** [review]
